### PR TITLE
TSM Feature

### PR DESCRIPTION
**Pull request by sc0ttkclark**
https://central.tri.be/issues/108891

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Sales.php:398](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/Tribe/Commerce/PayPal/Orders/Sales.php#L398)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$event_id' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:144](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/Tribe/Status/Abstract_Commerce.php#L144)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$sku' | [Abstract_Commerce.php:181](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/Tribe/Status/Abstract_Commerce.php#L181)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:62](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/admin-views/tpp-orders.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:95](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/admin-views/tpp-orders.php#L95)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:108](https://github.com/moderntribe/event-tickets/blob/c1a86845933d1dbbb09bb2b0e8a26a7435f9c3f7/src/admin-views/tpp-orders.php#L108)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Sales.php:398](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/Tribe/Commerce/PayPal/Orders/Sales.php#L398)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$event_id' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:144](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/Tribe/Status/Abstract_Commerce.php#L144)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$sku' | [Abstract_Commerce.php:181](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/Tribe/Status/Abstract_Commerce.php#L181)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:62](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/admin-views/tpp-orders.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:95](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/admin-views/tpp-orders.php#L95)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:108](https://github.com/moderntribe/event-tickets/blob/37c501cda72205fb6a6226a6419ad7d176175995/src/admin-views/tpp-orders.php#L108)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Sales.php:398](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/Tribe/Commerce/PayPal/Orders/Sales.php#L398)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$event_id' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:144](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/Tribe/Status/Abstract_Commerce.php#L144)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$sku' | [Abstract_Commerce.php:181](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/Tribe/Status/Abstract_Commerce.php#L181)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:62](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/admin-views/tpp-orders.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:95](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/admin-views/tpp-orders.php#L95)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:108](https://github.com/moderntribe/event-tickets/blob/acb00facbbda8d7fb6135dabd4bdb16f20d8fcaa/src/admin-views/tpp-orders.php#L108)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Sales.php:398](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/Tribe/Commerce/PayPal/Orders/Sales.php#L398)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$event_id' | [Abstract_Commerce.php:143](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/Tribe/Status/Abstract_Commerce.php#L143)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Abstract_Commerce.php:144](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/Tribe/Status/Abstract_Commerce.php#L144)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$sku' | [Abstract_Commerce.php:181](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/Tribe/Status/Abstract_Commerce.php#L181)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:62](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/admin-views/tpp-orders.php#L62)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_id' | [tpp-orders.php:72](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/admin-views/tpp-orders.php#L72)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:95](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/admin-views/tpp-orders.php#L95)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$order_overview' | [tpp-orders.php:108](https://github.com/moderntribe/event-tickets/blob/df80f7e0df31afad86bb5f57c8ab661586fc5f1d/src/admin-views/tpp-orders.php#L108)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Commerce/PayPal/Orders/Report.php [hunk 1]
```diff
@@ -250,18 +249,57 @@ public function orders_page_inside() {
...
+				);
+			}
+
+			//update ticket item counts by order status
+			$orders = $sales->get_all_orders_by_product_id( $ticket->ID );
+			foreach ( $orders as $key => $order ) {
+
+				if ( $order->get_status_label() && $order->get_item_quantity() ) {
+
+					if ( in_array( $order->get_status(), $complete_statuses ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Orders/Report.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/Commerce/PayPal/Orders/Report.php [hunk 2]
```diff
@@ -250,18 +249,57 @@ public function orders_page_inside() {
...
+						$tickets_sold[ $ticket->name ]['completed'] += $order->get_item_quantity();
+					}
 
+					if ( in_array( $order->get_status(), $incomplete_statuses ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Orders/Report.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/RSVP.php [hunk 3]
```diff
@@ -451,7 +469,7 @@ public function update_attendee_data( $data, $order_id, $event_id ) {
 
 		//check if changing status will cause rsvp to go over capacity
 		$previous_order_status = get_post_meta( $order_id, self::ATTENDEE_RSVP_KEY, true );
-		if ( tribe_is_truthy( $attendee_order_status ) && 'no' === $previous_order_status ) {
+		if ( tribe_is_truthy( $attendee_order_status ) && in_array( $previous_order_status, $this->get_statuses_by_action( 'count_not_going' ) ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/RSVP.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/RSVP.php [hunk 4]
```diff
@@ -625,7 +643,8 @@ public function send_tickets_email( $order_id, $event_id = null ) {
 		foreach ( $all_attendees as $single_attendee ) {
 			// Do not add those attendees/tickets marked as not attending (note that despite the name
 			// 'qr_ticket_id', this key is not QR code specific, it's simply the attendee post ID)
-			if ( 'yes' !== get_post_meta( $single_attendee['qr_ticket_id'], self::ATTENDEE_RSVP_KEY, true ) ) {
+			$going_status = get_post_meta( $single_attendee['qr_ticket_id'], self::ATTENDEE_RSVP_KEY, true );
+			if ( in_array( $going_status, $this->get_statuses_by_action( 'count_not_going' ) ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/RSVP.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/RSVP.php [hunk 5]
```diff
@@ -2142,11 +2162,10 @@ public function update_stock_from_attendees_page( $attendee_id ) {
 	 */
 	public function get_total_not_going( $event_id ) {
 
-		$not_going = 0;
-
+		$not_going     = 0;
 		foreach ( $this->get_attendees_array( $event_id ) as $attendee ) {
-			if ( 'no' === $attendee[ 'order_status' ] ) {
-				$not_going++;
+			if ( in_array( $attendee['order_status'], $this->get_statuses_by_action( 'count_not_going' ) ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/RSVP.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/Ticket_Object.php [hunk 6]
```diff
@@ -547,11 +547,12 @@ public function inventory() {
 			// Fetch the Attendees
 			$attendees = $this->provider->get_attendees_by_id( $this->ID );
 			$attendees_count = 0;
+			$not_going_arr = tribe( 'tickets.status' )->get_statuses_by_action( 'count_not_going', 'rsvp' );
 
 			// Loop on All the attendees, allowing for some filtering of which will be removed or not
 			foreach ( $attendees as $attendee ) {
 				// Prevent RSVP with Not Going Status to decrease Inventory
-				if ( 'rsvp' === $attendee['provider_slug'] && 'no' === $attendee['order_status'] ) {
+				if ( ! empty( $attendee['provider_slug'] ) && 'rsvp' === $attendee['provider_slug'] && in_array( $attendee[ 'order_status' ], $not_going_arr ) ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Ticket_Object.php]: Always set in_array() third param to true when comparing strings.

File: src/Tribe/Status/Abstract_Commerce.php [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+	 * Get Order Tooltip
+	 *
+	 * @since tbd
+	 *
+	 * @return string a string of html for the tooltip
+	 */
+	public function get_total_order_tooltip() {
+		ob_start();
+		?>
+		<div class="tribe-tooltip" aria-expanded="false">
```
Comment (sc0ttkclark) [inline src/Tribe/Status/Abstract_Commerce.php]: Any reason these types of functions aren't using views instead?

Comment (jesseeproductions) [reply src/Tribe/Status/Abstract_Commerce.php]: Such as an admin view?

Comment (sc0ttkclark) [reply src/Tribe/Status/Abstract_Commerce.php]: Right, just curious why all the markup is in the class instead of views/admin views.

**Pull request merged**